### PR TITLE
Fix OLM for 0.12.1

### PR DESCRIPTION
**Pull request by aliok**
Fix bad OLM metadata as per https://github.com/operator-framework/community-operators/pull/1164

### CODE REVIEW

File: deploy/olm-catalog/knative-kafka-operator/0.12.1/knative-kafka-operator.v0.12.1.clusterserviceversion.yaml
Comment (matzew) [inline deploy/olm-catalog/knative-kafka-operator/0.12.1/knative-kafka-operator.v0.12.1.clusterserviceversion.yaml:63]: @aliok all the removed bits here, are those now actually taken into account from the "applied" yamls (of the `KafkaChannel`) ?

**Comment by matzew** [comment]
/hold 

until is merged: https://github.com/operator-framework/community-operators/pull/1210

**Review by matzew — CHANGES_REQUESTED** [review]
Please add 

```
  minKubeVersion: 1.15.0
```

to the CSV - like done on the Operatorhub PR

File: deploy/olm-catalog/knative-kafka-operator/0.12.1/knative-kafka-operator.v0.12.1.clusterserviceversion.yaml
Comment (aliok) [reply deploy/olm-catalog/knative-kafka-operator/0.12.1/knative-kafka-operator.v0.12.1.clusterserviceversion.yaml:63]: yeah, the operator-sdk version I have I think has a bug that it considered everything under `/deploy` recursively and ended up having all the resources here in this file. 

These rules are actually for the things that are provisioned by the operator, not for the operator itself.

**Comment by aliok** [comment]
`minKubeVersion: 1.15.0` is now added. Same content as https://github.com/operator-framework/community-operators/pull/1210

cc @maschmid

**Comment by aliok** [comment]
The operator hub PR is merged @matzew 

Feel free to merge or wait for @maschmid 's review

**Comment by maschmid** [comment]
/lgtm

@matzew  can we unhold this now?

**Pull request merged**
